lib/std/mem.zig

```diff
@@ -635,13 +635,76 @@ test "lessThan" {
 /// Compares two slices and returns whether they are equal.
 pub fn eql(comptime T: type, a: []const T, b: []const T) bool {
     if (a.len != b.len) return false;
-    if (a.ptr == b.ptr) return true;
+    if (a.len == 0 or a.ptr == b.ptr) return true;
+    if (@sizeOf(T) == 0) return true;
+
+    if (std.meta.hasUniqueRepresentation(T)) return eqlBytes(std.mem.sliceAsBytes(a), std.mem.sliceAsBytes(b));
```

Comment:
Should this test to call over to `eqlBytes()` happen before the len/ptr checks above?  Since those are duplicated in the body of eqlBytes and I think this is a comptime check, vs. those are run-time checks.

Reply:
The duplicate checks are optimized out trivially, however when I do what I said in sno2's review, I don't mind changing the order. There's no reason to, but 🤷 why not.

Reply:
Huh, that caused some unexpected results to be honest 😄

Reply:
In your last commit, you again moved the comptime checks below the len/ptr checks. What's the reason for this?

Reply:
Whoops, thank you for noticing. I guess it was a conflict between two commits I squashed. Will fix now :heart: